It seems gnome-shell consistently crashes when doing certain actions, e.g.
opening the settings from the top panel, when running in unaccelerated
environments, particularly qemu virtual machines, with llvmpipe. The
crash happens by gnome-shell getting killed.

It was found that inhibiting the KMS thread works around the problem, so
do that for now, at least as long as this is unfixed.

diff --git a/src/backends/native/meta-renderer-native.c b/src/backends/native/meta-renderer-native.c
index e6c653e..cf0ee1a 100644
--- a/src/backends/native/meta-renderer-native.c
+++ b/src/backends/native/meta-renderer-native.c
@@ -2177,6 +2177,12 @@ meta_renderer_native_initable_init (GInitable     *initable,
       flags = meta_kms_device_get_flags (kms_device);
       renderer_native->has_addfb2 = !!(flags & META_KMS_DEVICE_FLAG_HAS_ADDFB2);
 
+      /* if we're not accelerated, inhibit kms thread in order to avoid getting
+       * killed randomly e.g. in qemu virtual machines as that's quite annoying
+       */
+      if (!gpu_kms_is_hardware_rendering (renderer_native, renderer_native->primary_gpu_kms))
+        meta_kms_inhibit_kernel_thread (meta_kms_device_get_kms (kms_device));
+
       kms_modifiers_debug_env = g_getenv ("MUTTER_DEBUG_USE_KMS_MODIFIERS");
       if (kms_modifiers_debug_env)
         {
